File: src/backend/Application/UltimateGoalAggregate/Commands/UpdateUltimateGoalInformation/UpdateUltimateGoalInformationCommand.cs
using InternetSafetyPlan.Application.Base;
using MediatR;

namespace InternetSafetyPlan.Application.UltimateGoalAggregate.Commands;

public record UpdateUltimateGoalInformationCommand(Guid UltimateGoalId, string Name, string? Description = null) : ICommand<Unit>;

File: src/frontend/src/library/ultimateGoalAggregate/commands/UpdateUltimateGoalInformationCommand.ts
export class UpdateUltimateGoalInformationCommand {
  public ultimateGoalId: string;
  public name: string;
  public description: string | null;

  constructor(ultimateGoalId: string, name: string, description: string) {
    this.ultimateGoalId = ultimateGoalId;
    this.name = name;
    this.description = description;
  }
}